packages/date-picker/src/vaadin-date-picker-mixin.js

```diff
@@ -927,15 +972,14 @@ export const DatePickerMixin = (subclass) =>
         this.__showOthers();
         this.__showOthers = null;
       }
-
       window.removeEventListener('scroll', this._boundOnScroll, true);
 
       this.__commitParsedOrFocusedDate();
 
       if (this._nativeInput && this._nativeInput.selectionStart) {
         this._nativeInput.selectionStart = this._nativeInput.selectionEnd;
       }
-      // No need to revalidate the value after `_selectedDateChanged`
+      // No need to revalidate the value after it has been just committed.
       // Needed in case the value was not changed: open and close dropdown,
       // especially on outside click. On Esc key press, do not validate.
       if (!this.value && !this._keyboardActive) {
```

Comment:
**note:** The validation can now be triggered twice on `blur` when it coincides with `change`. I considered it not to be a serious issue since (1) the Flow components will no longer rely on client-side validation timing (2) `validate()` calls themselves are pretty cheap in web-components.